airbyte-integrations/connectors/source-hubspot/source_hubspot/streams.py

```diff
@@ -1676,6 +1676,98 @@ class Contacts(CRMSearchStream):
     scopes = {"crm.objects.contacts.read"}
 
 
+class ContactsMergedAudit(CRMSearchStream):
+    entity = "contact"
+    last_modified_field = "lastmodifieddate"
+    associations = []
+    data_field = "results"
+    primary_key = "vid-to-merge"
+    scopes = {"crm.objects.contacts.read"}
+
+    @property
+    def url_search(self):
+        return f"/crm/v3/objects/{self.entity}/search"
+
+    @property
+    def url(self):
+        """Get batch Contacts Endpoint, API v1
+        Is used to get merge-audits field for all contacts with merge history
+        Note: CRM API V3 do not return this field
+        Docs: https://legacydocs.hubspot.com/docs/methods/contacts/get_batch_by_vid
+        """
+        return "/contacts/v1/contact/vids/batch/"
+
+    def __init__(self, **kwargs):
+        """Override Stream init function
+        To use read_records from CRMSearchStream with Search endpoint only, we define _state value
+        """
+        super().__init__(**kwargs)
+        self._sync_mode = "incremental"
+        self._state = self.state if self.state else self._start_date
+
+    def get_json_schema(self) -> Mapping[str, Any]:
+        """Override get_json_schema defined in Stream class
+        Final object does not have properties field
+        We return JSON schema as defined in :
+        source_hubspot/schemas/contacts_merged_audit.json
+        """
+        return super(Stream, self).get_json_schema()
+
+    def _get_search_payload(self, next_page_token: Mapping[str, Any] = None):
+        """Create payload for CRM Search API
+        Filter contacts with hs_merged_object_ids is not null
+        and last_modified_field > state
+        """
+        payload = {
+            "filters": [
+                {"value": int(self._state.timestamp() * 1000), "propertyName": self.last_modified_field, "operator": "GTE"},
+                {"propertyName": "hs_merged_object_ids", "operator": "HAS_PROPERTY"},
+            ],
+            "sorts": [{"propertyName": self.last_modified_field, "direction": "ASCENDING"}],
+            "limit": 100,
+        }
+        if next_page_token:
+            payload.update(next_page_token["payload"])
+
+        return payload
+
+    @retry_connection_handler(max_tries=5, factor=5)
+    @retry_after_handler(fixed_retry_after=1, max_tries=3)
+    def _get_contact_merge_audit(
```

Comment:
Why we should send get request instead of use CDK in-build parent-child stream logic?

Reply:
Not sure to understand your question.

Do you mean : why do we need to do this get request ?

Reply:
Which function from `CRMSearchStream` or `Stream` should I use ?

Reply:
> Why we should send get request instead of use CDK in-build parent-child stream logic?
Which function from CRMSearchStream or Stream should I use ?

Reply:
> Why we should send get request instead of use CDK in-build parent-child stream logic?

Which function from CRMSearchStream or Stream should I use ?

Reply:
@lazebnyi what I understand is that you are asking me to leverage the `Contacts` stream as it's done [here](https://discuss.airbyte.io/t/implementing-nested-stream-using-full-refresh-parent/3084/4)

Problem is that we will parse all contacts to then call the API V1 instead of filtering with additional filter : 

```python
{"propertyName": "hs_merged_object_ids", "operator": "HAS_PROPERTY"}
```

Reply:
Any update on this @lazebnyi ? 😊

Reply:
Yes, I'm talking about that implementation. You can create helper stream to call API v1.

Reply:
Okey will do this 👍